lib/src/models/types/parameter-serialization-strategies.ts

```diff
@@ -0,0 +1,53 @@
+export interface ParamSerializationStrategy {
+  query?: QueryParamSerializationStrategy;
+}
+
+export interface QueryParamSerializationStrategy {
+  array?: QueryParamArrayStrategy;
+}
+
+/** Supported serialization strategies for arrays in query parameters */
+export type QueryParamArrayStrategy =
```

Comment:
This may be better represented as an `enum` and referenced in the `switch/case` below

Reply:
As discussed offline, `enum`s would require a more awkward contract syntax.